### PR TITLE
transfer models page to next

### PR DESCRIPTION
**Pull request by Adelle-Pitsas**
- Transfers models page to next
   - Adds use-client to layout and wraps layout in `STACProvider`
- Moved the page into a new route group called (modelsMap) since it has a map but isn't the platform map

### CODE REVIEW

File: app/(modelsMap)/models/page.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { DehydratedPlatforms } from "Features/ERDDAP/hooks/DehydrateComponent"
+import { ModelingPage } from "./modeling"
+
+export default async function ModelingIndexPage() {
+  return (
+    <DehydratedPlatforms>
```
Comment (abkfenris) [inline app/(modelsMap)/models/page.tsx:6]: Do we need `<DehydratedPlatforms>` here? We may want to dehydrate STAC data to speed up catalog loading at some point, but I don't think we have any platform data in this route.

I think `modeling.tsx` could be renamed to `page.tsx` with the exports tweaked.

File: src/components/NavBar/index.tsx [hunk 2]
```diff
@@ -60,13 +60,9 @@ const NeracoosNavBar = () => {
...
-
-            {!isMariners ? (
-              <NavItem>
-                <NavLink href={paths.models}>Model Viewer</NavLink>
-              </NavItem>
-            ) : null}
-
+            <NavItem>
+              <NavLink href={paths.models}>Model Viewer</NavLink>
+            </NavItem>
```
Comment (abkfenris) [inline src/components/NavBar/index.tsx:65]: Right now the model viewer (and maybe some of the servers behind it) aren't quite ready for full public consumption, so we still want to check if the site is at mariners.neracoos.org and hide it from the menu if that is the case. 

Probably the better route (so we avoid a flash as things get rehydrated and update), would be to not show it by default, and only show it if `!window.location.href.includes("://mariners.neracoos.org")`.

**Review by abkfenris — CHANGES_REQUESTED** [review]
Pretty close. I think we have an unneeded layer of hydration happening, and we need to hide the menu item on the public facing site.

File: app/(modelsMap)/models/page.tsx (hunk 1 above)
Comment (Adelle-Pitsas) [reply app/(modelsMap)/models/page.tsx:6]: Great--Took out `<DehydratedPlatforms>` and moved the main `<ModelingPage/>` components into `Page.tsx` to keep auxiliary components out of the main route.

**Review by abkfenris — APPROVED** [review]
Feel free to merge when you're happy

**Pull request merged**